src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -55,21 +116,91 @@ struct AggregateFunctionSumKahanData
     T sum{};
     T compensation{};
 
-    void add(T value)
+    template <typename Value>
+    void ALWAYS_INLINE addImpl(Value value, T & out_sum, T & out_compensation)
+    {
+        auto compensated_value = value - out_compensation;
+        auto new_sum = out_sum + compensated_value;
+        out_compensation = (new_sum - out_sum) - compensated_value;
+        out_sum = new_sum;
+    }
+
+    void ALWAYS_INLINE add(T value)
+    {
+        addImpl(value, sum, compensation);
+    }
+
+    /// Vectorized version
+    template <typename Value>
+    void NO_INLINE addMany(const Value * __restrict ptr, size_t count)
+    {
+        constexpr size_t unroll_count = 4;
```

Comment:
It is a little bit strange that for usual sum we used `128 / sizeof(T);`, but now it's 4.
Comment?